### PR TITLE
Add support for reconciler delete feature

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:
- Undeploy command now uses the reconciler delete feature
- Removed all custom deletion logic
- Refactor deploy/undeploy to no longer manage reconciler interaction
- Encapsulated all reconciler interaction in the deploy pkg

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/12272
https://github.com/kyma-incubator/reconciler/issues/500

### CODE REVIEW

**Comment by clebs** [comment]
Needs to be merged after https://github.com/kyma-incubator/reconciler/issues/333

**Review by jeremyharisch — COMMENTED** [review]
The `keep-crds` flag does not have a function when your PR is merged. Please consider removing it, or give the user a hint that this function is currently not implemented.

**Review by skhalash — COMMENTED** [review]
Left some comments about things to improve

File: internal/deploy/component/list.go [hunk 1]
```diff
@@ -24,6 +28,23 @@ type Definition struct {
...
+		filePath, err := resolve.File(componentsFile, filepath.Join(ws.WorkspaceDir, "tmp"))
+		if err != nil {
+			return List{}, err
+		}
+		return FromFile(filePath)
+	}
+
+	return FromFile(path.Join(ws.InstallationResourceDir, "components.yaml"))
+}
+
```
[the remark this answers is not in the document]
Comment (clebs) [reply internal/deploy/component/list.go]: I am also not so fond of the 3 args where not all at the same time make sense, but the dependency to the `reconciler/chart` is the main reason for this after being able to reuse this in `deploy` and `undeploy` commands.

My goal was to remove any dependency to the reconciler on the `cmd/` packages and confine it to `pkg/deploy`.

File: cmd/kyma/undeploy/cmd.go [hunk 2]
```diff
@@ -77,313 +80,97 @@ func (cmd *command) Run() error {
...
+		return errors.Errorf("Kyma undeployment failed.")
 	}
 
-	if err := cmd.removeCustomResourcesFinalizers(); err != nil {
-		return err
+	if recoResult.GetResult() == model.ClusterStatusDeleted {
+		unDeployStep.Success()
 	}
-
+	unDeployStep.Successf("Kyma undeployed successfully!")
```
[the remark this answers is not in the document]
Comment (clebs) [reply cmd/kyma/undeploy/cmd.go:140]: True, changeed this to be consistent

**Review by skhalash — APPROVED** [review]

**Review by jeremyharisch — APPROVED** [review]

File: docs/gen-docs/kyma_undeploy.md [hunk 3]
```diff
@@ -15,8 +15,23 @@ kyma undeploy [flags]
...
 
 ```bash
-      --keep-crds          Set --keep-crds=true to keep CRDs on clean-up
-      --timeout duration   Maximum time for the deletion (default 6m0s)
+      --component strings        Provide one or more components to undeploy (e.g. --component componentName@namespace)
+  -c, --components-file string   Path to the components file (default "$HOME/.kyma/sources/installation/resources/components.yaml" or ".kyma-sources/installation/resources/components.yaml")
+      --concurrency int          Set maximum number of workers to run simultaneously to deploy Kyma. (default 4)
+  -d, --domain string            Custom domain used for installation.
+  -p, --profile string           Kyma deployment profile. If not specified, Kyma uses its default configuration. The supported profiles are: evaluation, production.
+  -s, --source string            Installation source:
```
Comment (NHingerl) [inline docs/gen-docs/kyma_undeploy.md:23]:
```suggestion
  -s, --source string            Source of installation to be undeployed:
```

**Review by NHingerl — APPROVED** [review]

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**